tinygrad/shape/view.py

```diff
@@ -95,9 +96,12 @@ def reshape(self, new_shape: Tuple[sint, ...]) -> Optional[View]:
     # after the asserts, it's okay to check contiguous
     if self.contiguous: return View.create(new_shape)
 
-    # check if this is adding or removing 1s (only)
+    # check if this is adding or removing 1s with [x for x in self.shape if x != 1] == [x for x in new_shape if x != 1].
+    # extend the above to include cases like (3,4) --> (3,i) where we add or remove 1s and add/remove a "dummy" variable.
     # NOTE: this is optional, but removes most calls to (expensive!) merge_views (with mask, not optional)
-    if [x for x in self.shape if x != 1] == [x for x in new_shape if x != 1]:
+    diff_between_shapes = [(a,b) for a,b in zip_longest([x for x in self.shape if x != 1],[x for x in new_shape if x != 1]) if a != b]
```

Comment:
Huh? This is so complex looking!

Reply:
re read this and its complete garbage - sorry! 

Updated it again. 3rd time lucky 🤞